omero_marshal/encode/encoders/annotation.py

```diff
@@ -68,7 +68,7 @@ def encode(self, obj):
         if obj.isAnnotationLinksLoaded() and obj.sizeOfAnnotationLinks() > 0:
             annotations = list()
             for annotation_link in obj.copyAnnotationLinks():
-                annotation = annotation_link.child
+                annotation = annotation_link.getChild()
```

Comment:
Is the issue here that the wrapper doesn't handle the property access correctly?

Reply:
I'd have to reproduce, but I think it's when a non-wrapper is passed, that the property does not exist.

Reply:
At least for the base model objects I can't see this being required. Using `Roi` since you mentioned it offline:

```
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from omero.model import RoiI, RoiAnnotationLinkI, FileAnnotationI

In [2]: roi = RoiI(1, False)

In [3]: fa = FileAnnotationI(10, False)

In [4]: a_link = RoiAnnotationLinkI(100)

In [5]: a_link.setParent(roi)

In [6]: a_link.setChild(fa)

In [7]: a_link.getChild().getId()
Out[7]:
object #0 (::omero::RLong)
{
    _val = 10
}

In [8]: a_link.child.id
Out[8]:
object #0 (::omero::RLong)
{
    _val = 10
}
```

I looked into the wrappers and they don't override `copyAnnotationLinks()` as far as I can tell.

In conclusion, don't think we need this change.